### DISCUSSION TITLE
Why doesn't the sample plugin (LightPlugin) return the value as it shows in the example?

### DISCUSSION THREAD

**Discussion by bryantlin** [microsoft/semantic-kernel#6377] [Q&A] [answered]
Hi everyone, 

I am pretty new to Semantic Kernel and trying to run a small example (_LightPlugin_) from this page https://learn.microsoft.com/en-us/semantic-kernel/overview/?tabs=Csharp. 

I am stuck on the second answer from AI - in the example, it answers "_[Light is now on]
Assistant > I have turned on the lights for you._" but the answer I am getting is "_Assistant > I'm sorry but I am an AI language model and I do not have the ability to physically interact with your lights. However, if you are looking for a solution to control your lights, you may consider purchasing a smart home device such as a smart light bulb or a smart light switch that is compatible with your preferred voice assistant._"

The only difference from the example is I have:
"**var builder = Kernel.CreateBuilder().AddAzureOpenAIChatCompletion(DeploymentName, Endpoint, new DefaultAzureCredential());**"

In the example, it is "**var builder = Kernel.CreateBuilder().AddAzureOpenAIChatCompletion(modelId, endpoint, apiKey);**".

Does it have something to do with my Azure OpenAI instance/endpoint? Could you please what may be going on? 

Thank you for your help in advance!

**Comment by dmytrostruk** [comment, marked as answer]
Hi @bryantlin !
I don't think the problem is in difference in Azure OpenAI chat completion service registration. It looks like the model is not aware about your plugin to change the light state. If you use exactly the same code as on Microsoft Learn page, could you please share model id you are using in your testing? As far as I know, not all models support function calling functionality.

Here is a list of available models, their features and available regions:
https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models

**Comment by bryantlin** [comment]
Thanks @dmytrostruk! That should be the reason.

**Comment by dieter-arnold** [comment]
Hello everyone,

I have the same problem. I get the error message:

"I'm afraid I can't do that as I'm just a text-based AI language model. However, if you have a smart lighting system or a smart bulb connected to a smart hub, you can use its compatible app or voice assistant to turn on the light."

I use the model: gpt-35-turbo (0301)

Can anyone help me?

Thank you for your help in advance!

**Reply by dmytrostruk** [reply]
@dieter-arnold It looks like `gpt-35-turbo (0301)` doesn't support function calling. Here is a list of supported models:
https://platform.openai.com/docs/guides/function-calling/supported-models
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/72f27a03-978f-4e24-8b64-48587e9c62aa)

Could you please try to use another model from list above and see if the sample works correctly? Thanks in advance!

**Reply by dieter-arnold** [reply]
Thanks @dmytrostruk! That was the solution - I switched to a gpt-4 model.

Does anyone know when gpt4 will be available in the "Western Europe" region?
